### ISSUE TITLE
[Swift 5.6] Regression in `ClosureParameterPositionRule`

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#3845]
`ClosureParameterPositionRuleTests.testWithDefaultConfiguration()` test is failing with Xcode 13.3 beta 2.

**Comment by jpsim** [comment]
The following examples are no longer triggering a violation in Swift 5.6:

```swift
[1, 2].map {
    [weak ↓self] number in
    number + 1
}
```

```swift
f.completionHandler = {
    ↓thing in
    doStuff()
}
```

```swift
foo {
    [weak ↓self] in
    self?.bar()
}
```
